snorkel/utils/core.py

```diff
@@ -170,8 +170,10 @@ def filter_labels(
     """
     masks = []
     for label_name, filter_values in filter_dict.items():
-        if label_dict[label_name] is not None:
-            masks.append(_get_mask(label_dict[label_name], filter_values))
+        label_array: Optional[np.ndarray] = label_dict.get(label_name)
+        if label_array is not None:
+            # _get_mask requires not-null input
+            masks.append(_get_mask(label_array, filter_values))
```

Comment:
Mypy doesn't seem to be smartly enough to tell that `label_dict[label_name]` is not-null after the condition check `if label_dict[label_name] is not None`. It still treats `label_dict[label_name]` as `Optional[np.ndarray]` and pass it to `_get_mask` which has requirement on not-null input `def _get_mask(label_array: np.ndarray, filter_values: List[int])`. This causes mypy to error out.
By extracting `label_dict[label_name]` into another variable `label_array` the issue will go away.

Reply:
Does
```
assert label_dict[label_name] is not None
```
fix the mypy check? If not I think this should be fine (cc: @bhancock8 )

Reply:
@rsmith49 Tried that. It doesn't fix mypy error. Also not sure if it's a good idea to add run time check here to prevent the program from running if `label_dict[label_name]` is None

Reply:
Agreed, I meant after the if statement (sorry that wasn't clear). If mypy is having issues with dict values though, this change should be fine